CMakeLists.txt

```diff
@@ -66,16 +66,42 @@ endif()
 
 # Compiler dependent flags
 include (CheckCCompilerFlag)
-if(UNIX)
-    check_c_compiler_flag(-msse4.2 HAS_SSE42)
-    if(HAS_SSE42)
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse4.2")
-        add_definitions(-DHAS_SSE42)
+if(UNIX OR MINGW)
+    check_c_compiler_flag(-march=armv8-a+crc ARM_CRC)
+    if(ARM_CRC)
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=armv8-a+crc")
+    else()
+        check_c_compiler_flag(-msse2 HAS_SSE2)
+        if(HAS_SSE2)
+            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse2")
+            add_definitions(-DHAS_SSE2)
+        endif()
+
+        check_c_compiler_flag(-mssse3 HAS_SSSE3)
+        if(HAS_SSSE3)
+            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mssse3")
+            add_definitions(-DHAS_SSSE3)
+        endif()
+
+        check_c_compiler_flag(-msse4.2 HAS_SSE42)
+        if(HAS_SSE42)
+            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -msse4.2")
+            add_definitions(-DHAS_SSE42)
+        endif()
+
+        check_c_compiler_flag(-mpclmul HAS_PCLMUL)
+        if(HAS_PCLMUL)
+            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mpclmul")
+            add_definitions(-DHAS_PCLMUL)
+        endif()
     endif()
 elseif(MSVC)
     set(CMAKE_DEBUG_POSTFIX "d")
     add_definitions(-D_CRT_SECURE_NO_DEPRECATE)
     add_definitions(-D_CRT_NONSTDC_NO_DEPRECATE)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /wd4018")   # '<': signed/unsigned mismatch
```

Comment:
I wonder if we should instead fix those warnings?

Reply:
will check it again using MSVC. I think this only happens when we use MSVC.

Reply:
I can fix them myself really.

Reply:
Cool!